src/core/cgroup.c

```diff
@@ -1200,12 +1200,15 @@ static int cgroup_apply_devices(Unit *u) {
         return r;
 }
 
-static void set_io_weight(Unit *u, const char *controller, uint64_t weight) {
-        char buf[8+DECIMAL_STR_MAX(uint64_t)+1];
-        const char *p;
+static void set_io_weight(Unit *u, bool is_blkio, uint64_t weight) {
+        char buf[STRLEN("default ")+DECIMAL_STR_MAX(uint64_t)+1];
```

Comment:
I think it was decided not to manually add the +1 for nul, and that DECIMAL_STR_MAX accounts for it?

Reply:
It is for `\n`.

Reply:
ah, got it